src/oscar/apps/checkout/surcharges.py

```diff
@@ -0,0 +1,52 @@
+from decimal import Decimal as D
+
+from oscar.core import prices
+
+
+class Base(object):
```

Comment:
Please remove `object`. I also suggest we name this `BaseSurcharge` to avoid ambiguity when imported by projects.